libraries/AP_NMEA_Output/AP_NMEA_Output_config.h

```diff
@@ -0,0 +1,22 @@
+#include <AP_HAL/AP_HAL_Boards.h>
+
+// Needs SerialManager + (AHRS or GPS)
+
+#ifdef HAL_BUILD_AP_PERIPH
+    #define AP_NMEA_OUTPUT_LIBRARY_REQUIREMENTS (defined(HAL_GCS_ENABLED) && HAL_GCS_ENABLED && defined(HAL_PERIPH_ENABLE_GPS))
```

Comment:
Could we do all of this down in AP_Periph instead?

Reply:
yes, but why?

Reply:
To move it to periph, the only way would be in AP_Peirph.h, and anyone else touching NMEA_Output, to include AP_NMEA_Output_config.h, then check if NMEA is enabled before including the normal AP_NMEA_Output.h.

Isn't the purpose of the _config to do this kind of stuff?

Reply:
@peterbarker And here's what that looks like in Periph.h
```
#include <AP_NMEA_Output/AP_NMEA_Output_config.h>
#if HAL_NMEA_OUTPUT_ENABLED
    #if HAL_GCS_ENABLED && defined(HAL_PERIPH_ENABLE_GPS)
        #include <AP_NMEA_Output/AP_NMEA_Output.h>
    #else
        // Needs SerialManager + (AHRS or GPS)
        #error "AP_NMEA_Output requires Serial/GCS and either AHRS or GPS. Needs HAL_GCS_ENABLED and HAL_PERIPH_ENABLE_GPS"
    #endif
#endif // HAL_NMEA_OUTPUT_ENABLED
```

Reply:
ok, I gave in. I moved it to periph